src/bootstrap.php

```diff
@@ -0,0 +1,293 @@
+            case Response::HTTP_GONE:
+            case Response::HTTP_NOT_FOUND:
+                throw new HttpException($e->getResponse()->getStatusCode(), "$identifier does not exist", $e);
+        }
+
+        throw $e;
+    }
+
+    $relations = $app['elife.api_sdk']->articles()
+        ->getRelatedArticles($id)
+        ->sort(function (Article $a, Article $b) {
+            static $order = [
+                'retraction' => 1,
+                'correction' => 2,
+                'external-article' => 3,
+                'registered-report' => 4,
+                'replication-study' => 5,
+                'research-advance' => 6,
+                'scientific-correspondence' => 7,
+                'research-article' => 8,
+                'tools-resources' => 9,
+                'feature' => 10,
+                'insight' => 11,
+                'editorial' => 12,
+                'short-report' => 13,
+            ];
+
+            return $order[$a->getType()] <=> $order[$b->getType()];
+        });
+
+    $collections = $app['elife.api_sdk']->collections()
+        ->containing($article->getIdentifier())
+        ->slice(0, 100);
+
+    $podcastEpisodeChapters = $podcastEpisodes
+        ->reduce(function (Sequence $chapters, PodcastEpisode $episode) use ($article) {
+            foreach ($episode->getChapters() as $chapter) {
+                foreach ($chapter->getContent() as $content) {
+                    if ($article->getId() === $content->getId()) {
+                        $chapters = $chapters->append(new PodcastEpisodeChapterModel($episode, $chapter));
+                        continue 2;
+                    }
+                }
+            }
+
+            return $chapters;
+        }, new EmptySequence());
+
+    if ($article->getSubjects()->notEmpty()) {
+        $subject = $article->getSubjects()[0];
+
+        $mostRecentWithSubject = $app['elife.api_sdk']->search()
+            ->forType('correction', 'editorial', 'feature', 'insight', 'research-advance', 'research-article', 'retraction', 'registered-report', 'replication-study', 'scientific-correspondence', 'short-report', 'tools-resources')
+            ->sortBy('date')
+            ->forSubject($subject->getId())
+            ->slice(0, 5);
+    } else {
+        $mostRecentWithSubject = new EmptySequence();
+    }
+
+    $mostRecent = $app['elife.api_sdk']->search()
+        ->forType('research-advance', 'research-article', 'scientific-correspondence', 'short-report', 'tools-resources', 'replication-study')
+        ->sortBy('date')
+        ->slice(0, 5);
+
+    $recommendations = $relations;
+
+    $appendFirstThatDoesNotExist = function (Sequence $recommendations, Sequence $toInsert) : Sequence {
+        foreach ($toInsert as $item) {
+            foreach ($recommendations as $recommendation) {
+                if (
+                    get_class($item) === get_class($recommendation)
+                    &&
```

Comment:
isn't it more general to compare directly `get_class($item).'::'.$item->getIdentifier()`?